Automattic/wp-calypso
# Partner Program: Adding a referrer prop to sign up start tracks event

Related to https://github.com/Automattic/jetpack-manage/issues/48

## Proposed Changes

* This PR adds a referrer prop to the sign up start tracks event - `calypso_partner_portal_agency_signup_start` - in the `<AgencySignupForm>` component.
This referrer prop is named `formreferrer`, as the name `referrer` results in no updates to eventprops when viewing in Tracks live view, though the referrer prop is visible in Tracks Vigilante. Using `referrer` does seem to result in the `logreferrer` prop itself being  updated, so to prevent any mixups I've chosen to make sure 'referrer' is not in the prop name.
* Note - I originally had issues with the event double firing on each page refresh in both Tracks Vigilante and Live View. The prop would only be added on the second firing. I'm not too sure why - but stopping my build and restarting seemed to help.
So with this PR - the firing of `calypso_partner_portal_agency_signup_start` on page refresh (using the agency / manage sign up form here) results in an eventprop `formreferrer` value of `manage-lp`.

## Testing Instructions

* Open the agency sign up page while in development mode - `http://jetpack.cloud.localhost:3000/agency/signup` - you'll need a fresh WPcom account for this.
* You'll also need to apply this PR, or wait until it is merged: https://github.com/Automattic/wp-calypso/pull/83308
* With Tracks Vigilante or looking on Tracks live view, check for the `calypso_partner_portal_agency_signup_start` event to fire. If looking on Tracks live view, check event props and you should see the `formreferrer` with the `manage-lp` value. In Tracks Vigilante the `formreferrer` prop should be visible, with the `manage-lp` value.
